packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js

```diff
@@ -41,6 +56,14 @@ const DropdownItem = ({ className, children, isHovered, component: Component, is
       <Component
         {...additionalProps}
         className={css(isDisabled && styles.modifiers.disabled, isHovered && styles.modifiers.hover, className)}
+        onClick={event => {
+          if (Component === 'button') {
+            onClick && onClick(event);
```

Comment:
Why both onClick and onSelect?

Reply:
I see, onSelect is added in the Menu. Maybe remove from propTypes so it is only documented that user can add onClick to the item?

Reply:
Actually, nevermind, better would be to use context i think

Reply:
I think it would be best to create a context in Dropdown, and then pass the onSelect through that instead of props. Additionally, when you return the onSelect to the user, apart from the event maybe also return the index